From 5dc99237d49e73c27d3eca54f6ccd97d13f94de0 Mon Sep 17 00:00:00 2001
From: Marco Nelissen <marcone@google.com>
Date: Fri, 21 Oct 2016 09:41:00 -0700
Subject: Tremolo: fix ARM assembly code for decode_map type 3 case

Bug: 31647370
Test: ran CTS and manual PoC tests
Change-Id: If176cee6c267d9025d5c6af49c19354192f901a9
(cherry picked from commit 798ceb60518ebbc9eb2f5005a83cd7779eabb879)
---
 Tremolo/dpen.s | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Tremolo/dpen.s b/Tremolo/dpen.s
index 9bc6f18..7e4f1ed 100644
--- a/Tremolo/dpen.s
+++ b/Tremolo/dpen.s
@@ -362,7 +362,7 @@ dm3:
 	@ r12= mul
 	@ r14= add
 	LDR	r1, [r4,#44]		@ r1 = s->q_pack
-	LDR	r4, [r4,#52]		@ r4 = s->q_val
+	LDR	r4, [r4,#48]		@ r4 = s->q_val
 	CMP	r2,#8
 	MOV	r11,#0			@ r11= prev
 	MLA	r4,r1,r8,r4		@ r4 = ptr = s->q_val+entry*s->q_pack
-- 
cgit v0.12

